source/PreferencesPanel.cpp

```diff
@@ -178,6 +198,29 @@ bool PreferencesPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &comma
 		if(zones[latest].Value().KeyName() != Command::MENU.KeyName())
 			Command::SetKey(zones[latest].Value(), 0);
 	}
+	else if(key == 'i' && page == 'p')
+	{
+		page = 'i';
+		if(!downloadedInfo)
+		{
+			PluginHelper::Download(PLUGIN_LIST_URL.c_str(), (Files::Config() + "plugins.json").c_str());
+			ifstream pluginlistFile(Files::Config() + "plugins.json");
+			pluginInstallList = nlohmann::json::parse(pluginlistFile);
+			pluginlistFile.close();
+			pluginInstallPages = ((pluginInstallList.size() - (pluginInstallList.size() % MAX_PLUGIN_INSTALLS_PER_PAGE))
+				/ MAX_PLUGIN_INSTALLS_PER_PAGE) + (pluginInstallList.size() % MAX_PLUGIN_INSTALLS_PER_PAGE > 0);
+			downloadedInfo = true;
+		}
+	}
+	else if(key == 'i' && page == 'i' && selectedPluginInstall.url.size())
```

Comment:
It looks like it is possible to install the same plugin multiple times?

Reply:
did something about that, basically involves a set that contains the names of all active plugins that is then always guarded by a mutex